engine/src/hopeit/server/logger.py

```diff
@@ -38,7 +38,8 @@ class EventLoggerWrapper:
     def __init__(self, logger: logging.Logger):
         self.logger = logger
 
-    def debug(self, msg, *args, **kwargs) -> None:
+    def debug(self, context: EventContext, msg, *args, **kwargs) -> None:
+        _enrich_extra(kwargs, context, msg)
```

Comment:
We don't want to even do this call when loglevel > DEBUG, that's why enriching extra arguments is not supported in debug logging.

Reply:
Maybe checking current log level if it does not affect performance?

Reply:
Or capture the exception and throw a proper message indicating context/extra arguments are not supported.

Reply:
Removing the context avoids any check

Reply:
Debug logger processor only in debug level